### PR TITLE
update tests with new auto_opt api

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

updates tests missed here https://github.com/PyTorchLightning/pytorch-lightning/pull/5169

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]
correct

File: tests/core/test_lightning_optimizer.py [hunk 1]
```diff
@@ -89,6 +89,9 @@ def test_lightning_optimizer_manual_optimization(mock_sgd_step, mock_adam_step,
     Test that the user can use our LightningOptimizer. Not recommended for now.
     """
     class TestModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.automatic_optimization = False
```
Comment (justusschock) [inline tests/core/test_lightning_optimizer.py:94]:
```suggestion
            self._automatic_optimization = False
```

File: tests/core/test_lightning_optimizer.py [hunk 2]
```diff
@@ -141,6 +140,9 @@ def test_lightning_optimizer_manual_optimization_and_accumulated_gradients(mock_
     Test that the user can use our LightningOptimizer. Not recommended.
     """
     class TestModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.automatic_optimization = False
```
Comment (justusschock) [inline tests/core/test_lightning_optimizer.py:145]:
```suggestion
            self._automatic_optimization = False
```

File: tests/plugins/test_ddp_sequential_plugin.py [hunk 3]
```diff
@@ -149,6 +149,7 @@ class SequentialModelRPCManual(LightningModule):
     def __init__(self):
         super().__init__()
         self.sequential_module = nn.Sequential(torch.nn.Linear(32, 32), nn.ReLU(), nn.Linear(32, 2))
+        self.automatic_optimization = False
```
Comment (justusschock) [inline tests/plugins/test_ddp_sequential_plugin.py:152]:
```suggestion
        self._automatic_optimization = False
```

File: tests/plugins/test_ddp_sequential_plugin.py [hunk 4]
```diff
@@ -195,19 +196,14 @@ def val_dataloader(self):
...
 
-    @property
-    def automatic_optimization(self) -> bool:
-        return False
-
 
 class SequentialModelRPCAutomatic(SequentialModelRPCManual):
+    def __init__(self):
+        super().__init__()
+        self.automatic_optimization = False
```
Comment (justusschock) [inline tests/plugins/test_ddp_sequential_plugin.py:203]:
```suggestion
        self._automatic_optimization = False
```

File: tests/trainer/dynamic_args/test_multiple_optimizers.py [hunk 5]
```diff
@@ -68,6 +69,10 @@ def test_multiple_optimizers_manual(tmpdir):
     Tests that only training_step can be used
     """
     class TestModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.automatic_optimization = False
```
Comment (justusschock) [inline tests/trainer/dynamic_args/test_multiple_optimizers.py:74]:
```suggestion
            self._automatic_optimization = False
```

File: tests/trainer/optimization/test_manual_optimization.py [hunk 6]
```diff
@@ -99,6 +99,10 @@ def test_multiple_optimizers_manual_return(tmpdir):
     Tests that only training_step can be used
     """
     class TestModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.automatic_optimization = False
```
Comment (justusschock) [inline tests/trainer/optimization/test_manual_optimization.py:104]:
```suggestion
            self._automatic_optimization = False
```

File: tests/trainer/optimization/test_manual_optimization.py [hunk 7]
```diff
@@ -166,6 +166,10 @@ def test_multiple_optimizers_manual_return_and_log(tmpdir):
...
```
Comment (justusschock) [inline tests/trainer/optimization/test_manual_optimization.py:171]:
```suggestion
            self._automatic_optimization = False
```

File: tests/trainer/optimization/test_manual_optimization.py [hunk 8]
```diff
@@ -239,6 +239,10 @@ def test_multiple_optimizers_manual_native_amp(tmpdir):
...
```
Comment (justusschock) [inline tests/trainer/optimization/test_manual_optimization.py:244]:
```suggestion
            self._automatic_optimization = False
```

File: tests/trainer/optimization/test_manual_optimization.py [hunk 9]
```diff
@@ -308,6 +308,10 @@ def test_multiple_optimizers_manual_apex(tmpdir):
...
```
Comment (justusschock) [inline tests/trainer/optimization/test_manual_optimization.py:313]:
```suggestion
            self._automatic_optimization = False
```

File: tests/trainer/optimization/test_manual_optimization.py [hunk 10]
```diff
@@ -381,6 +381,10 @@ class ManualOptimizationExtendedModel(BoringModel):
     called = collections.defaultdict(int)
     detach = False
 
+    def __init__(self):
+        super().__init__()
+        self.automatic_optimization = False
```
Comment (justusschock) [inline tests/trainer/optimization/test_manual_optimization.py:386]:
```suggestion
        self._automatic_optimization = False
```

File: tests/trainer/optimization/test_manual_optimization.py [hunk 11]
```diff
@@ -503,6 +503,10 @@ class ExtendedModel(BoringModel):
         called = collections.defaultdict(int)
         detach = False
 
+        def __init__(self):
+            super().__init__()
+            self.automatic_optimization = False
```
Comment (justusschock) [inline tests/trainer/optimization/test_manual_optimization.py:508]:
```suggestion
            self._automatic_optimization = False
```

File: tests/trainer/optimization/test_manual_optimization.py [hunk 12]
```diff
@@ -587,6 +587,10 @@ class TestModel(BoringModel):
 
         called = False
 
+        def __init__(self):
+            super().__init__()
+            self.automatic_optimization = False
```
Comment (justusschock) [inline tests/trainer/optimization/test_manual_optimization.py:592]:
```suggestion
            self._automatic_optimization = False
```

File: tests/trainer/optimization/test_manual_optimization.py [hunk 13]
```diff
@@ -666,6 +666,10 @@ class TestModel(BoringModel):
 
         _losses = []
 
+        def __init__(self):
+            super().__init__()
+            self.automatic_optimization = False
```
Comment (justusschock) [inline tests/trainer/optimization/test_manual_optimization.py:671]:
```suggestion
            self._automatic_optimization = False
```

File: tests/trainer/optimization/test_manual_optimization.py [hunk 14]
```diff
@@ -743,6 +743,10 @@ def test_step_with_optimizer_closure_and_accumulated_grad(tmpdir):
     os.environ['PL_DEV_DEBUG'] = '1'
 
     class TestModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.automatic_optimization = False
```
Comment (justusschock) [inline tests/trainer/optimization/test_manual_optimization.py:748]:
```suggestion
            self._automatic_optimization = False
```

File: tests/trainer/optimization/test_manual_optimization.py [hunk 15]
```diff
@@ -806,6 +806,10 @@ def test_step_with_optimizer_closure_and_extra_arguments(step_mock, tmpdir):
...
```
Comment (justusschock) [inline tests/trainer/optimization/test_manual_optimization.py:811]:
```suggestion
            self._automatic_optimization = False
```

File: tests/trainer/optimization/test_manual_optimization.py [hunk 16]
```diff
@@ -863,6 +863,10 @@ def test_step_with_optimizer_closure_with_different_frequencies(mock_sgd_step, m
...
```
Comment (justusschock) [inline tests/trainer/optimization/test_manual_optimization.py:868]:
```suggestion
            self._automatic_optimization = False
```

File: tests/trainer/optimization/test_manual_optimization.py [hunk 17]
```diff
@@ -944,6 +944,9 @@ def test_step_with_optimizer_closure_with_different_frequencies_ddp(mock_sgd_ste
...
```
Comment (justusschock) [inline tests/trainer/optimization/test_manual_optimization.py:949]:
```suggestion
            self._automatic_optimization = False
```

**Review by justusschock — APPROVED** [review]
@rohitgr7 I added some suggestions, since the way you implemented it, you overwrite the property, but this way we just change the value used by the property.

**Comment by rohitgr7** [comment]
@justusschock there's a setter for that.
https://github.com/PyTorchLightning/pytorch-lightning/blob/652df1886abc63820d956aef7a468ff4cf1bea7f/pytorch_lightning/core/lightning.py#L169-L171

**Review by carmocca — APPROVED** [review]

File: tests/plugins/test_ddp_sequential_plugin.py (hunk 4 above)
Comment (rohitgr7) [inline tests/plugins/test_ddp_sequential_plugin.py:203]:
```suggestion
        self.automatic_optimization = True
```
